### PR TITLE
ci: upgrade to use Go 1.16.7

### PR DESCRIPTION
**Pull request by rboyer**

### CODE REVIEW

File: .changelog/10856.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+build: upgrade to use Go 1.16.7
```
Comment (mikemorris) [inline .changelog/10856.txt:2]:
```suggestion
ci: upgrade to use Go 1.16.7
```

**Review by mikemorris — APPROVED** [review]

File: .changelog/10856.txt (hunk 1 above)
Comment (rboyer) [reply .changelog/10856.txt]: fixed

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/428198.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit e0d1e2689cf692a34087e77aa399c5dab3233bc2 onto `release/1.10.x` succeeded!

**Pull request merged**